src/command.rs

```diff
@@ -0,0 +1,306 @@
+    let mut manifest = Manifest::read().await?;
+
+    manifest
+        .dependencies
+        .push(Dependency::new(repository.to_owned(), package, version));
+
+    manifest.write().await
+}
+
+/// Removes a dependency from this project
+pub async fn remove(package: PackageName) -> eyre::Result<()> {
+    let mut manifest = Manifest::read().await?;
+
+    let dependency = manifest
+        .dependencies
+        .iter()
+        .find(|d| d.package == package)
+        .wrap_err(eyre::eyre!(
+            "Unable to remove unknown dependency {package:?}"
+        ))?
+        .to_owned();
+
+    manifest.dependencies.retain(|d| *d != dependency);
+
+    PackageStore::uninstall(&dependency.package).await.ok();
+
+    manifest.write().await
+}
+
+/// Packages the api and writes it to the filesystem
+pub async fn package(directory: String, dry_run: bool) -> eyre::Result<()> {
+    let package = PackageStore::release()
+        .await
+        .wrap_err("Failed to create release")?;
+
+    let path = Path::new(&directory).join(format!(
+        "{}-{}.tgz",
+        package.manifest.package.name, package.manifest.package.version
+    ));
+
+    if !dry_run {
+        std::fs::write(path, package.tgz).wrap_err("failed to write package to filesystem")?;
+    }
+
+    Ok(())
+}
+
+/// Publishs the api package to the registry
+pub async fn publish(
+    credentials: Credentials,
+    repository: String,
+    allow_dirty: bool,
+    dry_run: bool,
+) -> eyre::Result<()> {
+    if let Ok(repository) = git2::Repository::discover(Path::new(".")) {
+        let statuses = repository
+            .statuses(None)
+            .wrap_err("Failed to get git status")?;
+
+        if !allow_dirty && !statuses.is_empty() {
+            tracing::error!("{} files in the working directory contain changes that were not yet committed into git:\n", statuses.len());
+
+            statuses
+                .iter()
+                .for_each(|s| tracing::error!("{}", s.path().unwrap_or_default()));
+
+            tracing::error!("\nTo proceed with publishing despite the uncommitted changes, pass the `--allow-dirty` flag\n");
+
+            eyre::bail!("Unable to publish a dirty git repository");
+        }
+    }
+
+    let artifactory = {
+        let Some(artifactory) = credentials.artifactory else {
+            eyre::bail!(
+                "Unable to publish package to artifactory, please login using `buffrs login`"
+            );
+        };
+
+        Artifactory::from(artifactory)
+    };
+
+    let package = PackageStore::release()
+        .await
+        .wrap_err("Failed to create release")?;
+
+    if dry_run {
+        tracing::warn!(":: aborting upload due to dry run");
+        return Ok(());
+    }
+
+    artifactory.publish(package, repository).await?;
+
+    Ok(())
+}
+
+/// Installs dependencies
+pub async fn install(credentials: Credentials) -> eyre::Result<()> {
+    let artifactory = {
+        let Some(artifactory) = credentials.artifactory else {
```

Comment:
credentials here should be optional - e.g. Artifactory has anonymous mode auth

Reply:
I think #56 covers this.